src/openms/source/KERNEL/ConsensusMap.cpp

```diff
@@ -730,6 +731,96 @@ OPENMS_THREAD_CRITICAL(oms_log)
     return true;
   }
 
+  void ConsensusMap::split(std::vector<FeatureMap>& fmaps, ConsensusMap::SplitMeta mode) const
+  {
+    // Prepare output vector
+    fmaps.clear();
+    Size numbr_exps = column_description_.size();
+    fmaps.resize(numbr_exps);
+
+    // Check for Isobaric Analyzer
+    bool iso_analyze = true;
+    auto cm_dp = this->getDataProcessing(); // get a copy to avoid calling .begin() and .end() on two different temporaries
+    if (all_of(cm_dp.begin(), cm_dp.end(), [](const OpenMS::DataProcessing& dp)
+                                           { return (dp.getProcessingActions().count(DataProcessing::QUANTITATION) == 0); }))
+    {
+      iso_analyze = false;
+    }
+
+    for (const auto& cf : *this)
+    {
+      // Create new Features from FeatureHandles
+      std::vector<BaseFeature> new_feats(numbr_exps);
+      for (const FeatureHandle& fh : cf.getFeatures())
+      {
+        BaseFeature feat(fh);
+        new_feats[fh.getMapIndex()] = feat;
+      }
+
+      // Add PeptideIdentifications to ...
+      for (const PeptideIdentification& pep_id : cf.getPeptideIdentifications())
+      {
+
+        if (iso_analyze)
+        {
+          // ... the first Feature.
+          new_feats[0].getPeptideIdentifications().push_back(pep_id);
```

Comment:
this (i.e. `fmaps[0].back()....` is now dangerous, since the current CF might not have a BaseFeature for Map #0 (because its from IsoAnalyzer, it SHOULD have, but please check it by remembering the smallest map index for this CF and throw exception for anything other than 0).